src/graph/entities/attribute_set.c

```diff
@@ -28,10 +28,10 @@ static bool _AttributeSet_Remove
 	Attribute_ID attr_id
 ) {
 	AttributeSet _set = *set;
-	int attr_count = _set->attr_count;
+	const uint16_t attr_count = _set->attr_count;
 
 	// locate attribute position
-	for(int i = 0; i < attr_count; i++) {
+	for (uint16_t i = 0; i < attr_count; ++i) {
```

Comment:
it seems like you prefer using prefix operations `++I` over `I++` if there's no real good reason for doing so please switch to suffix `I++`